### PR TITLE
Add storage checks

### PR DESCRIPTION
**Pull request by amityadav0**
- Add storage checks in swap functions

### CODE REVIEW

File: nearswap/src/lib.rs [hunk 1]
```diff
@@ -227,6 +230,8 @@ impl NearSwap {
         let (mut p, tokens_out) = self._price_n2t_in(&token, ynear);
         assert_min_buy(tokens_out, min_tokens);
         self._swap_n2t(&mut p, ynear, &token, tokens_out);
+        d.update_storage(start_storage);
+        self.set_deposit(&user, &d);
```
Comment (robert-zaremba) [inline nearswap/src/lib.rs:234]: let's put lines 223,224,233 and 234 on one helper function. The deposit (`d`) is not used in between.

Comment (amityadav0) [reply nearswap/src/lib.rs:234]: ok

File: nearswap/src/lib.rs [hunk 2]
```diff
@@ -363,9 +377,17 @@ impl NearSwap {
             tokens_out,
             near_swapped,
         );
+        self.storage_check(start_storage);
         return tokens_in_to_pay.into();
     }
 
+    pub fn storage_check(&mut self, storage: StorageUsage) {
```
Comment (robert-zaremba) [inline nearswap/src/lib.rs:384]: Let's use this interface
```
    pub fn unsafe_storage_check(&mut self, start_storage: StorageUsage) {
```

Comment (robert-zaremba) [reply nearswap/src/lib.rs:384]: 1. Must be a private function
2. add doc string to explain that this is unsafe when we have other deposit instance around this function call, which can overwrite changes here.

**Review by robert-zaremba — APPROVED** [review]

File: nearswap/src/lib.rs (hunk 2 above)
Comment (amityadav0) [reply nearswap/src/lib.rs:384]: ok

**Pull request merged**